### PR TITLE
[docs] describe console logging of the build process

### LINKED ISSUE #18778: [docs] add console.log instructions to debugging the build process
## Description

A suggestion from documentation feedback asked if there was a way to `console.log()` output from the build process. Reviewing the [Debugging the Build Process](https://www.gatsbyjs.org/docs/debugging-the-build-process/#vs-code-debugger-auto-config) doc shows a detailed guide on setting up debugging for VSCode or the Chrome developer tools. However, at its simplest, knowledge that you can `console.log` output from different steps in the build process is nice to be aware of and is much more approachable to those familiar with the browser's `console.log` than setting up new configurations for more sophisticated debugging.

## What to Do

I propose we add a section into the guide with a title like "Node.js console logging" that could mention Node.js' built-in console. 

This section could provide code snippets of places like in [Gatsby Node APIs](https://www.gatsbyjs.org/docs/node-apis/) or in a `gatsby-config` where you can log information and see it show up in logs that could help someone unfamiliar with Gatsby internals to see where their code is being executed in the build process. It could also help to link off to resources like [Overview of the Gatsby Build Process](https://www.gatsbyjs.org/docs/overview-of-the-gatsby-build-process/) for more information.

### PR DESCRIPTION
**Pull request by kriswep**
## Description

Document using `console` to debug Gatsbys' build process.
Used the established example to debug and added references to Node.js console docs as well as to the build process overview.

Feedback welcome...

## Related Issues
Fixes #18778

### CODE REVIEW

File: docs/docs/debugging-the-build-process.md [hunk 1]
```diff
@@ -40,6 +41,33 @@ TypeError: Cannot read property 'internal' of undefined
     D:/dev/blog-v2/gatsby-node.js:6:12
 ```
 
+## Debugging with Node.js' built-in console
+
+One of the fastest ways to gain insight into Gatsby's build process is using the `console` functionality [built into Node.js](https://nodejs.org/en/knowledge/getting-started/the-console-module/). This works similar to how you might be used to in the browser.
+
+Adding a console.log statement in our sample from above will print the variable into your terminal. There you might notice that `args` contains a lower-cased node variable.
```
Comment (marcysutton) [inline docs/docs/debugging-the-build-process.md]:
```suggestion
Adding a `console.log` statement in the sample from above will print the variable into your terminal. There you might notice that `args` contains a lower-cased node variable.
```

Updating to adhere to the Gatsby Style Guide: https://www.gatsbyjs.org/contributing/gatsby-style-guide/#use-you-as-the-pronoun

File: docs/docs/debugging-the-build-process.md [hunk 2]
```diff
@@ -40,6 +41,33 @@ TypeError: Cannot read property 'internal' of undefined
...
+    createNodeField({
+      name: `slug`,
+      node,
+      value,
+    })
+  }
+}
+```
+
+For a refresher what is part of the build process check out the differences between [build and runtime](/docs/overview-of-the-gatsby-build-process#build-time-vs-runtime). Generally spoken, Node.js is responsible for building Gatsby pages and therefore its' built-in objects like `console` can be used.
```
Comment (marcysutton) [inline docs/docs/debugging-the-build-process.md]:
```suggestion
To read more about Gatsby's build process, check out the differences between [build and runtime](/docs/overview-of-the-gatsby-build-process#build-time-vs-runtime). Generally spoken, Node.js is responsible for building Gatsby pages and therefore its' built-in objects like `console` can be used at build time. At client-side [runtime](/docs/glossary#runtime), the browser's `console.log` API will add messages to the developer tools console. 
```

File: docs/docs/debugging-the-build-process.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ Gatsby's `build` and `develop` steps run as a Node.js application which you can
 
 In this guide you will learn how to debug some code using:
 
+- [Debugging with Node.js' built-in console](#debugging-with-nodejs-built-in-console)
```
Comment (marcysutton) [inline docs/docs/debugging-the-build-process.md]: One more thing: this hard-coded table of contents can be removed as there's an automatic one on the page now.

**Comment by kriswep** [comment]
Just added the suggested changes and rephrased the `let's` sentences.

**Comment by laurieontech** [comment]
Looks great! Going to merge this now. Thanks for the PR, think this will help a lot with debugging efforts.

File: docs/docs/debugging-the-build-process.md [hunk 4]
```diff
@@ -40,6 +36,33 @@ TypeError: Cannot read property 'internal' of undefined
...
+
+One of the fastest ways to gain insight into Gatsby's build process is using the `console` functionality [built into Node.js](https://nodejs.org/en/knowledge/getting-started/the-console-module/). This works similar to how you might be used to in the browser.
+
+Adding a `console.log` statement in the sample from above will print the variable into your terminal. There you might notice that `args` contains a lower-cased node variable.
+
+```diff:title=gatsby-node.js
+const { createFilePath } = require("gatsby-source-filesystem")
+
+exports.onCreateNode = args => {
++ console.log(args)
```
Comment (muescha) [inline docs/docs/debugging-the-build-process.md]: maybe it is better here to use `// highlight-line` so the syntax highlighter we can set to javascript?

Comment (laurieontech) [reply docs/docs/debugging-the-build-process.md]: That's a good catch. This was recently merged, so feel free to open a new PR to address this.

**Pull request merged**